### PR TITLE
Change "neopixel" references to use RGB LED naming

### PR DESCRIPTION
**Pull request by SuGlider**
## Description of Change

Internal request to remove all references to Trade Mark terms (neopixel).
Documentation, commentaries, symbols and functions were modified.

## Tests scenarios

CI Only.

## Related links

None - Internal.

### CODE REVIEW

**Comment by SuGlider** [comment]
@me-no-dev | @rftafas | @VojtechBartoska 

PTAL.

**Comment by me-no-dev** [comment]
@SuGlider my only "comment" is: Should we maybe keep the `neopixelWrite` in pins_arduino for adafruit boards?

@ladyada this is the new PR. Please have a look

**Comment by tonhuisman** [comment]
Hmm, how much confusion will this cause for people trying to connect/use 4-pin RGB leds with their boards (and as mounted on some boards instead of a NeoPixel), only to find they are supposed to use ~NeoPixel~ 'Serial Controlled RGB(W)' leds... 🤔

**Comment by me-no-dev** [comment]
@tonhuisman these functions are generally to be used with board LEDs and we do plan to add support for 4 pin RGB and DotStar LEDs

**Comment by ladyada** [comment]
reviewed - we're OK with any variant using the word 'NEOPIXEL' in the board def .... you may want to contact those board makers to give them the alert that a built in pin name has changed :)

fyi DotStar is also a trademark

**Comment by me-no-dev** [comment]
> fyi DotStar is also a trademark

It's a household name at this point, just like NeoPixel :) (props to you)

**Comment by SuGlider** [comment]
> @SuGlider my only "comment" is: Should we maybe keep the `neopixelWrite` in pins_arduino for adafruit boards?
> 
> @ladyada this is the new PR. Please have a look

Based on recent note from @ladyada:

Given that the `neopixelWrite()` is now called `rgbLedWrite()`, all commentaries have changed for all boards, including Adafruit.
I think that we should keep the PR because those commentaries were added by myself, not by Adafruit.

I also think that, for Arduino Core 3.0.x,  we should remove the `#define neopixelWrite rgbLedWrite` declaration and use a `neopixelWrite()` function, marking it as deprecated, instead.
Latter in Arduino Core 3.1, we should just definitively remove it.

Please let me know your considerations.

File: docs/en/api/rmt.rst [hunk 1]
```diff
@@ -17,7 +17,7 @@ To get started with RMT, you can try:
 RMT Write Neo Pixel
```
Comment (lucasssvaz) [inline docs/en/api/rmt.rst:17]:
```suggestion
RMT Write RGB LED
```

**Comment by SuGlider** [comment]
@me-no-dev @P-R-O-C-H-Y @lucasssvaz - Ready for review.

**Review by P-R-O-C-H-Y — APPROVED** [review]

**Review by me-no-dev — APPROVED** [review]

**Review by lucasssvaz — APPROVED** [review]
Works fine

**Pull request merged**